src/compiler/typescript_to_rust/azle_generate/src/candid_type/record/mod.rs

```diff
@@ -16,25 +19,41 @@ use super::errors::WrongEnclosedType;
 
 impl SourceMapped<'_, TsTypeAliasDecl> {
     pub fn to_record(&self) -> Result<Option<Record>, Vec<Error>> {
-        self.process_ts_type_ref("Record", |type_ref| {
-            let (type_params, members) =
+        self.process_ts_type_ref(&self.symbol_table.record, |type_ref| {
+            let (type_params, record_right_half) =
```

Comment:
rename to "right_hand_side" or "members"
actually call it "type_ref"
type Coolboy = Record<{}>